### DISCUSSION TITLE
Routing : First Solution Strategy

### DISCUSSION THREAD

**Discussion by kk2491** [google/or-tools#2372] [Routing (and legacy CP) questions]
**What version of OR-tools and what language are you using?**
Version: master.
Language: Python

**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
Routing Solver

**What operating system (Linux, Windows, ...) and version?**
Ubuntu 16.04

I have been going through the [Travelling Salesman Problem ](https://developers.google.com/optimization/routing/tsp#main), and found that "First Solution Strategy" which creates initial route for the solver. 
Could you please elaborate   
why this initial route is important?   
Is it really required to create this initial route ?  
Does the initial route is used as benchmark to get the optimized route?

Thank you,
KK

**Comment by haakonhr** [comment]
The initial route is the starting point for the local search, which always maintains a feasible solution. You can skip that step by using "ALL_UNPERFORMED" as your first solution strategy and adding disjunctions for all nodes (so that the solution where every node is unvisited is a feasible solution)
